app/models/user.rb

```diff
@@ -179,6 +179,12 @@ def related_hmis_user(data_source)
     Hmis::User.find(id)&.tap { |u| u.update(hmis_data_source_id: data_source.id) }
   end
 
+  def any_hmis_access?
+    return false unless HmisEnforcement.hmis_enabled?
+
+    Hmis::UserGroupMember.where(user_id: id).exists? # belongs to any HMIS user groups
```

Comment:
maybe?
```suggestion
    return @any_hmis_access unless @any_hmis_access.nil?
    return false unless HmisEnforcement.hmis_enabled?

    @any_hmis_access  = Hmis::UserGroupMember.where(user_id: id).exists? # belongs to any HMIS user groups
    @@any_hmis_access
```

Reply:
Not sure that gains us anything at the moment ^^^ given that we only call it once, but if we expect other HMIS links in the menu, it would mean only executing the query once.

Reply:
Thanks, ok. I was worried that the nav template code would execute multiple times as you navigate around. I am gonna go ahead and merge. I think when we have a multi-HMIS install we will need to revisit this linking anyway .

Reply:
Sounds good.  It will get executed on every page load at the moment, but hopefully that's a well optimized query.  If not, we should work on that, or cache for a few minutes.

Reply:
it should be well optimized